### ISSUE TITLE
Add sonification for the flux meter

### ISSUE THREAD

**Issue by jbphet** [phetsims/greenhouse-effect#185]
The flux meter is now working on the 2nd and 3rd screens, so it would be a good time to start working on the sound design for it.  We discussed this in today's sound design meeting, and here are a few notes:

- Probably the most important of the four values portrayed by the flux meter is the arrow that represents the amount of IR energy that is returning to the Earth, which is the downward pointing red arrow
- The second most important is the amount of IR moving in the upwards direction
- The sunlight is, overall, less important than the IR
- In some cases in the Layer Model screen, oscillations of the arrows can occur.  This is probably an emergent property of the model, but it is definitely _not_ something that we want to draw attention to

Assigning to @Ashton-Morris for some initial brainstorming and prototyping.

**Comment by Ashton-Morris** [comment]
I am adding this mockup here that we thought was a good direction to start in.

[- GHE_ Energy_Flux_Top_and_Bottom_Mockup.mp4](https://drive.google.com/file/d/19762bubnL04QrFb7clvEWQ04qhKwlnKW/view?usp=sharing)

**Comment by jbphet** [comment]
@Ashton-Morris and I discussed this in the 9/20/2022 sound design meeting, and he has some recollections about the feedback he got from this mockup.  He'll write those up in this issue, then create some sounds that I can try to integrate.

**Comment by Ashton-Morris** [comment]
[Adding some files. ](https://drive.google.com/drive/folders/19bEE-mjkSfpZBq5C6TtlKjP4yJa5zr3C?usp=sharing)

- GHE_Energy_Flux_Top_and_Top_IR.wav
- GHE_Energy_Flux_Top_and_Top_IR.mp3
- GHE_Energy_Flux_Top_and_Bottom_IR_B.wav
- GHE_Energy_Flux_Top_and_Bottom_IR_B.mp3
- GHE_Energy_Flux_Top_and_Bottom_IR_A.wav
- GHE_Energy_Flux_Top_and_Bottom_IR_A.mp3

In the mapping "GHE_Energy_Flux_Top_and_Top_IR" just goes up in down in pitch based off of the min/max of the Top red IR parameter. You could also copy the file and some implementation mapping from Gravity Force Labs Basics for Force.

For the Bottom IR, the mapping has it fade between the audio files "GHE_Energy_Flux_Top_and_Bottom_IR" at 0% and "GHE_Energy_Flux_Top_and_Bottom_IR_B." at 100% of its total range.

[Here is a new mockup video. ](https://drive.google.com/file/d/1Rxspwc4QLRYhJRe_uIR1Z69sX8hDMgnv/view?usp=sharing)

**Comment by jbphet** [comment]
Note: I've downloaded `GHE_Energy_Flux_Top_and_Top_IR` and renamed to `irFluxUp`.

**Comment by jbphet** [comment]
I've improved the fade in/out algorithm for the sound generation for IR in the up direction and added the pitch adjustment.  At this point, it works reasonably well on the Photons screen, but is too active on the Layer Model screen.  I'll need to refine the thresholds and algorithms to make this work better.  Once this is done, I'll move on to making the sounds work for the IR flux in the downward direction.

**Comment by jbphet** [comment]
I've added sound for the downward direction.  It uses a base sound and a low-pass filter to create the effect that I believe I'm hearing in the mockup.  I think this is now ready for review with the sound design team.

**Comment by jbphet** [comment]
At the sound design meeting yesterday we realized that @Ashton-Morris had intended to cross fade between two sounds for the downward direction IR instead of using a bandpass filter.  I'll fix this up for the next review.

**Comment by jbphet** [comment]
The sound design just reviewed the behavior in our weekly meeting.  I think we all felt like it's not perfect, but it's pretty good. The biggest problem is that it's a bit hard to tease out the distinction between the sound that represent IR moving up and IR moving down, but this is known to be a tricky issue with sonification in general.  So, the bottom line is that we're going to go with what we've got, at least for now.

By the way, we intentionally made a decision a while back to only sonify the IR flux and not the visible flux, since the behavior of the IR is much more related to the primary learning goals of the sim.

I think that's it for this issue.  Closing.
